### ISSUE TITLE
It seems there is a bug in the grammar for Story 1

### ISSUE THREAD

**Issue by glebzlat** [gvanrossum/pegen#256]
In the First story's toy grammar `if_statement` is going after `assignment` and `expr`, but in this case `if` will be recognized as `NAME` and the rest will not be parsed.

```diff
class ToyParser(Parser):

    def statement(self):
+       if i := self.if_statement():
+           return i
        if a := self.assignment():
            return a
        if e := self.expr():
            return e
-       if i := self.if_statement():
-           return i
        return None
```

**Comment by gvanrossum** [comment]
Thank you! This is indeed a bug. I will apply your fix and add a small test.

**Comment by gvanrossum** [comment]
Fixed by 2210f733afd0c5450a5a5ab6412d22dccade822b.
